controller/search.go

```diff
@@ -243,3 +245,43 @@ func (c *SearchController) Users(ctx *app.UsersSearchContext) error {
 
 	return ctx.OK(&response)
 }
+
+// Iterate over the WI list and read parent IDs
+// Fetch and load Parent WI in the included list
+func (c *SearchController) enrichWorkItemList(ctx *app.ShowSearchContext, res *app.SearchWorkItemList) {
+	// Need a map of string to UUID to prevent duplicates in place.
+	visitedIDs := map[string]uuid.UUID{}
+	for _, wi := range res.Data {
+		cond := wi.Relationships != nil && wi.Relationships.Parent != nil && wi.Relationships.Parent.Data != nil
+		if cond {
+			parentIDStr := *wi.Relationships.Parent.Data.ID
+			if _, exist := visitedIDs[parentIDStr]; !exist {
+				id, err := uuid.FromString(parentIDStr)
+				if err != nil {
+					log.Error(ctx, map[string]interface{}{
+						"wi_id": parentIDStr,
+						"err":   err,
+					}, "Parent ID is invalid UUID: %s", parentIDStr)
+				}
+				visitedIDs[parentIDStr] = id
+			}
+		}
+	}
+	for _, parentID := range visitedIDs {
```

Comment:
This is creating as many database transactions as visited IDs. Is it possible to run one query and retrieve all results?

Reply:
The conversion could be in a separate loop.

Reply:
Thanks @baijum , done in https://github.com/fabric8-services/fabric8-wit/pull/1592/commits/7635485b66848d121953a0c903290016b3c4db97
Please verify